TestParallelNewApp: inconsistent cluster.yaml and info.yaml

I've seen this a couple of times in CI now, [here](https://github.com/canonical/go-dqlite/actions/runs/4532508816/jobs/7983983908) is an affected run:

```
2023-03-27T13:46:55.3725590Z     --- FAIL: TestParallelNewApp/run-81 (0.00s)
2023-03-27T13:46:55.3725780Z         app_test.go:1056: 
2023-03-27T13:46:55.3726067Z             	Error Trace:	app_test.go:1056
2023-03-27T13:46:55.3726351Z             	Error:      	Received unexpected error:
2023-03-27T13:46:55.3726826Z             	            	inconsistent info.yaml and cluster.yaml
2023-03-27T13:46:55.3727160Z             	Test:       	TestParallelNewApp/run-81
```

--- a/app/app_test.go
+++ b/app/app_test.go
@@ -1056,8 +1056,8 @@ func TestParallelNewApp(t *testing.T) {
 			)
 			require.NoError(tt, err)
 			defer func() {
-				_ = os.RemoveAll(tmpDir)
 				_ = dqApp.Close()
+				_ = os.RemoveAll(tmpDir)
 			}()
 		})
 	}

